### PR TITLE
minor: fix broken links

### PR DESCRIPTION
**Pull request by pbludov**
Current report: https://checkstyle.org/linkcheck.html

No changes in the code.
No changes in the docs.

Only URI fixes for HTTP 301, 302, 403 and 404.
Also, obsolete suppressions for linkcheck have been removed.

New linkcheck report:
https://pbludov.github.io/site/linkcheck.html

### Not fixed:
* `File` type property of `Checker`, see
https://pbludov.github.io/site/config.html#Checker_Properties
The link to https://pbludov.github.io/site/property_types.html#file is invalid, there is no such type as `File`.
* Links to `.dtd` files are marked as invalid, but they to exist
* Links to **non-public** fields and methods are invalid on javadoc side, but they works fine within IDE.

These errors are complicated and cannot be fixed as part of a "minor" update. I will start new issues for them.

### CODE REVIEW

File: .github/ISSUE_TEMPLATE.md [hunk 1]
```diff
@@ -20,6 +20,6 @@ Describe what you expect in detail.
 --------------
 
 Still not clear ???
-see example - https://checkstyle.org/report_issue.html#How_to_report_a_bug
```
Comment (pbludov) [inline .github/ISSUE_TEMPLATE.md]: link fixed

File: pom.xml [hunk 2]
```diff
@@ -1514,12 +1514,12 @@
           <excludedLinks>
             <excludedLink>reports/google-style/guava</excludedLink>
             <excludedLink>reports/javadoc/openjdk8</excludedLink>
+            <!-- posting to the mailing list "checkstyle-announce" is private -->
```
Comment (pbludov) [inline pom.xml]: [maling-lists.html](https://checkstyle.org/mailing-lists.html) contains unclickable link. The `post` element is required, but can't be filled with the right value.

File: src/main/java/com/puppycrawl/tools/checkstyle/checks/annotation/MissingDeprecatedCheck.java [hunk 3]
```diff
@@ -68,7 +68,8 @@
  * Property {@code violateExecutionOnNonTightHtml} - If turned on, will
  * print violations if the Javadoc being examined by this check violates the
  * tight html rules defined at
- * <a href="writingjavadocchecks.html#Tight-HTML_rules">Tight-HTML Rules</a>.
```
Comment (pbludov) [inline src/main/java/com/puppycrawl/tools/checkstyle/checks/annotation/MissingDeprecatedCheck.java]: relative link. This is a bug left from xdocs-to-javadocs  align process.

File: src/main/java/com/puppycrawl/tools/checkstyle/checks/coding/CovariantEqualsCheck.java [hunk 4]
```diff
@@ -46,8 +46,8 @@
  * that are equal using covariant enum method, and not equal when compared normally.
  * </p>
  * <p>
- * Inspired by <a href="https://cs.nyu.edu/~lharris/papers/findbugsPaper.pdf">
```
Comment (pbludov) [inline src/main/java/com/puppycrawl/tools/checkstyle/checks/coding/CovariantEqualsCheck.java]: The author's home page has changed. Old home page is inaccessible. New home page contains updated version of the article. This is why the section number is changed also.

File: src/main/java/com/puppycrawl/tools/checkstyle/checks/coding/SimplifyBooleanReturnCheck.java [hunk 5]
```diff
@@ -43,7 +43,7 @@
  * </pre>
  * <p>
  * The idea for this Check has been shamelessly stolen from the equivalent
- * <a href="http://pmd.sourceforge.net">PMD</a> rule.
```
Comment (pbludov) [inline src/main/java/com/puppycrawl/tools/checkstyle/checks/coding/SimplifyBooleanReturnCheck.java]: PMD home page has changed.

File: src/main/java/com/puppycrawl/tools/checkstyle/filters/SuppressWithPlainTextCommentFilter.java [hunk 6]
```diff
@@ -65,7 +65,7 @@
  * </p>
  * <p>
  * Properties {@code offCommentFormat} and {@code onCommentFormat} must have equal
- * <a href="https://docs.oracle.com/javase/11/docs/api/java/util/regex/Matcher.html#groupCount()">
```
Comment (pbludov) [inline src/main/java/com/puppycrawl/tools/checkstyle/filters/SuppressWithPlainTextCommentFilter.java]: invalid link. This is a bug left from xdocs-to-javadocs align process.

File: src/xdocs/checks.xml [hunk 7]
```diff
@@ -371,7 +371,7 @@
             by the format property.</td>
           </tr>
           <tr>
-            <td><a href="config_naming.html#InvalidJavadocPosition">
```
Comment (pbludov) [inline src/xdocs/checks.xml]: mistype.

File: src/xdocs/config_coding.xml [hunk 8]
```diff
@@ -3683,7 +3683,7 @@ System.out.println(s2 instanceof Square); //true
             <tr>
               <td>treatTryResourcesAsStatement</td>
               <td>Enable resources processing.</td>
-              <td><a href="property_types.html#Boolean">Boolean</a></td>
```
Comment (pbludov) [inline src/xdocs/config_coding.xml]: mistype.

File: src/xdocs/eclipse.xml [hunk 9]
```diff
@@ -24,7 +24,7 @@
 
     <section name="Download Eclipse">
       <p>
-        Please visit Eclipse IDE <a href="https://eclipse.org"> official web site</a>,
```
Comment (pbludov) [inline src/xdocs/eclipse.xml]: HTTP 301 Moved Permanently to https://www.eclipse.org/

File: src/xdocs/writingfilters.xml [hunk 10]
```diff
@@ -106,7 +106,7 @@ public class FilesFilter
 
     <section name="Declare check's external resource locations">
       <p>
-        See <a href="writingchecks.html#Declare_checks_external_resource_locations">
```
Comment (pbludov) [inline src/xdocs/writingfilters.xml]: fixed broken link.

File: src/main/java/com/puppycrawl/tools/checkstyle/checks/javadoc/NonEmptyAtclauseDescriptionCheck.java [hunk 11]
```diff
@@ -44,7 +44,7 @@
  * RETURN_LITERAL</a>,
  * <a href="https://checkstyle.org/apidocs/com/puppycrawl/tools/checkstyle/api/JavadocTokenTypes.html#THROWS_LITERAL">
  * THROWS_LITERAL</a>,
- * <a href="https://checkstyle.org/apidocs/com/puppycrawl/tools/checkstyle/api/JavadocTokenTypes.html#TEXCEPTION_LITERAL">
```
Comment (pbludov) [inline src/main/java/com/puppycrawl/tools/checkstyle/checks/javadoc/NonEmptyAtclauseDescriptionCheck.java]: mistype

File: src/xdocs/contributing.xml [hunk 12]
```diff
@@ -67,7 +67,7 @@
 
       <p>
         To report issue please follow our best practices page - <a href="report_issue.html">
-          How to report an issue ?</a> (<a href="report_issue.html#How_to_report_a_bug">
+          How to report an issue ?</a> (<a href="report_issue.html#How_to_report_a_bug.3F">
```
Comment (strkkk) [inline src/xdocs/contributing.xml]: May be remove whitespace before question mark as well?
Same in line 71

Comment (pbludov) [reply src/xdocs/contributing.xml]: done

**Comment by romani** [comment]
Do you know why CI did report problems with links ? This should be detected automatically and notify us, as all other validation

**Comment by romani** [comment]
Codeship changed integration with GitHub https://blog.codeship.com/updated-github-integration/ .
I have migrated our project now, so it should start to work.

**Comment by romani** [comment]
Ok, now we have master build problem https://app.codeship.com/projects/124310/builds/1dec320f-1160-4ac4-8c2b-5a4562b8e86e?step=parallel_.ci%2Frun-link-check-plugin.sh , please investigate

**Comment by romani** [comment]
problem was:
`2019-12-08 21:00:33
 system <td><i><a class="externalLink" href="https://www.oracle.com/technetwork/java/javaee/overview/index.html">https://www.oracle.com/technetwork/java/javaee/overview/index.html</a>: 301 Moved Permanently</i></td></tr></table></td></tr>
2019-12-08 21:00:33
 system ------------ grep of linkcheck.html--END`

I used https://wheregoes.com/retracer.php to get destination

**Pull request merged**